code/modules/art/statues.dm

```diff
@@ -573,23 +573,27 @@ Moving interrupts
 	var/static/list/plane_whitelist = list(FLOAT_PLANE, GAME_PLANE, FLOOR_PLANE)
 
 	/// Ideally we'd have knowledge what we're removing but i'd have to be done on target appearance retrieval
-	var/list/overlays_to_remove = list()
+	var/list/overlays_to_keep = list()
 	for(var/mutable_appearance/special_overlay as anything in content_ma.overlays)
 		var/mutable_appearance/real = new()
 		real.appearance = special_overlay
 		if(PLANE_TO_TRUE(real.plane) in plane_whitelist)
-			continue
-		overlays_to_remove += real
-	content_ma.overlays -= overlays_to_remove
+			content_ma.overlays -= real
+			overlays_to_keep += real
+		else
+			content_ma.overlays -= real
+	content_ma.overlays = overlays_to_keep
```

Comment:
I am not really sure why this works exactly, this should have basically the same effect

Reply:
Its a mystery

Reply:
I like mysteries, but I don't like code I don't understand

Reply:
The only idea i have is that modifying mutable layers doesn't work as well as overwriting it entirely

Reply:
Or setting a mutable appearance by adding/removing is weird and not handled correctly

Reply:
it is possible that copying appearances as we do here is failing to copy some component of them and causing the removal to fail

Reply:
I'll need proof of that tho, would need to stick some print statements of overlay counts before/after the removal, alongside the amount of overlays we EXPECT to remove

Reply:
![Screenshot 2024-02-09 233528](https://github.com/tgstation/tgstation/assets/16868239/7460566a-4b94-4d4d-8c01-95389367f9d5)
As it so happens I was actually looking into statue code for another idea i had and had the whitelist checker modified for this new item and had an issue with layering as i was testing, but it seems that the unmodified statues do have more overylays than the corrected whitelist version

Reply:
![image](https://github.com/tgstation/tgstation/assets/16868239/deead729-7731-49e8-9b83-5cfd6ca7d8f0)
and If I were to simplify it by using the unmodified statue whitelist and the new whitelist check with a lantern turned on, it still includes the light in the unmodified version for both over and underlays.
![image](https://github.com/tgstation/tgstation/assets/16868239/2ef01f87-6089-402a-978e-48b7a1e1d461)